pkg/startup/startup_linux.go

```diff
@@ -100,3 +103,41 @@ func ipv6Supported() bool {
 func ensureNetworkForOS(ctx context.Context, client client.Interface, nodeName string) error {
 	return nil
 }
+
+func MonitorIPAddressSubnets() {
+	autoDetectPollingInterval := DEFAULT_AUTODETECT_POLL_INTERVAL
+	if os.Getenv("AUTODETECT_POLL_INTERVAL") != "" {
+		autoDetectPollingInterval, _ = time.ParseDuration(os.Getenv("AUTODETECT_POLL_INTERVAL"))
+	}
+
+	// Add a subscription to get updated if there is any change to interface addresses.
+	addrUpdate := make(chan netlink.AddrUpdate)
+	done := make(chan struct{})
+	if err := netlink.AddrSubscribe(addrUpdate, done); err != nil {
+		log.WithError(err).Error("Failed to subscribe to network interface updates")
+	}
+
+	ctx := context.Background()
+	_, cli := calicoclient.CreateClient()
+	nodeName := determineNodeName()
+	node := getNode(ctx, cli, nodeName)
+
+	for {
+		select {
+		case <-time.After(autoDetectPollingInterval):
```

Comment:
case statements don't automatically fall through in golang, so you'll need an explicit `fallthrough` keyword here to make it do so.

Also probably worth a log stating that this resync has been triggered by a periodic timer, rather than an interface update.